kafka/src/main/java/org/openconnectors/kafka/KafkaSource.java

```diff
@@ -0,0 +1,127 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Properties;
+
+import static org.openconnectors.config.ConfigUtils.verifyExists;
+
+/**
+ * Simple Kafka Source to emit strng messages from a topic
+ */
+public class KafkaSource implements PushSourceConnector<String> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaSource.class);
+
+    private Consumer<String, String> consumer;
+    private Properties props = new Properties();
+    private String topic;
+    private Boolean autoCommitEnabled;
+
+    private Object waitObject;
+    private java.util.function.Consumer<Collection<String>> consumeFunction;
+
+    @Override
+    public void open(Config config) throws Exception {
+
+        verifyExists(config, ConfigKeys.KAFKA_SINK_TOPIC);
+        verifyExists(config, ConfigKeys.KAFKA_SINK_BOOTSTRAP_SERVERS);
+        verifyExists(config, ConfigKeys.KAFKA_SOURCE_GROUP_ID);
+        verifyExists(config, ConfigKeys.KAFKA_SOURCE_FETCH_MIN_BYTES);
+        verifyExists(config, ConfigKeys.KAFKA_SOURCE_AUTO_COMMIT_INTERVAL_MS);
+        verifyExists(config, ConfigKeys.KAFKA_SOURCE_SESSION_TIMEOUT_MS);
+
+        topic = config.getString("kafka.source.topic");
+        autoCommitEnabled = config.getBoolean("kafka.source.auto_commit_enabled");
+
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, config.getString(ConfigKeys.KAFKA_SOURCE_BOOTSTRAP_SERVERS));
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, config.getString(ConfigKeys.KAFKA_SOURCE_GROUP_ID));
+        props.put(ConsumerConfig.FETCH_MIN_BYTES_CONFIG, config.getString(ConfigKeys.KAFKA_SOURCE_FETCH_MIN_BYTES));
+        props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, config.getString(ConfigKeys.KAFKA_SOURCE_AUTO_COMMIT_INTERVAL_MS));
+        props.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, config.getString(ConfigKeys.KAFKA_SOURCE_SESSION_TIMEOUT_MS));
+
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+
+    }
+
+    @Override
+    public void close() throws IOException {
+        LOG.info("Stopping kafka source");
+        if(consumer != null) {
+            consumer.close();
+        }
+        LOG.info("Kafka source stopped.");
+    }
+
+    public void start() throws Exception {
+        Thread runnerThread = new Thread(() -> {
+            LOG.info("Starting kafka source");
+            consumer = new KafkaConsumer<>(props);
+            consumer.subscribe(Arrays.asList(topic));
+            LOG.info("Kafka source started.");
+            ConsumerRecords<String, String> records;
+            while(true){
+                records = consumer.poll(1000);
```

Comment:
is it better to construct one collection rather than called consumeFunction repeatedly?

Reply:
really depends on the consume function but I will batch it for now.